ruby/rspec/style_guide.md

```diff
@@ -0,0 +1,645 @@
+end
+```
+
+### The One Expectation
+
+Use only one expectation per example. There are very few scenarios where two
+or more expectations in a single `it` block should be used. So, general rule
+of thumb is one expectation per `it` block.
+
+#### Bad Example
+
+```ruby
+describe ArticlesController do
+  #...
+
+  describe 'GET new' do
+    it 'assigns new article and renders the new article template' do
+      get :new
+      expect(assigns[:article]).to be_a(Article)
+      expect(response).to render_template :new
+    end
+  end
+
+  # ...
+end
+```
+
+#### Good Example
+
+```ruby
+describe ArticlesController do
+  #...
+
+  describe 'GET new' do
+    it 'assigns a new article' do
+      get :new
+      expect(assigns[:article]).to be_a(Article)
+    end
+
+    it 'renders the new article template' do
+      get :new
+      expect(response).to render_template :new
+    end
+  end
+end
+```
+
+### Context Cases
+
+`context` blocks should pretty much always have an opposite negative case. It
+should actually be a strong code smell if there is a single context (without a
+matching negative case) that it needs refactoring, or may have no purpose.
+
+#### Bad Example
+
+```ruby
+# This is a case where refactoring is the correct choice
+describe '#attributes' do
+  context 'the returned hash' do
+    it 'includes the display name' do
+      # ...
+    end
+
+    it 'includes the creation time' do
+      # ...
+    end
+  end
+end
+
+# This is a case where the negative case needs to be tested, but wasn't
+describe '#attributes' do
+  context 'when display name is present' do
+    before do
+      subject.display_name = 'something'
+    end
+
+    it 'includes the display name' do
+      # ...
+    end
+  end
+end
+```
+
+#### Good Example
+
+```ruby
+# Refactored
+describe '#attributes' do
+  subject { create(:article) }
+
+  expect(subject.attributes).to include subject.display_name
+  expect(subject.attributes).to include subject.created_at
+end
+
+# Added the negative case
+describe '#attributes' do
+  context 'when display name is present' do
+    before do
+      subject.display_name = 'something'
+    end
+
+    it 'includes the display name' do
+      # ...
+    end
+  end
+
+  context 'when display name is not present' do
+    before do
+      subject.display_name = nil
+    end
+
+    it 'does not include the display name' do
+      # ...
+    end
+  end
+end
+```
+
+### `context` descriptions
+
+`context` block descriptions should always start with 'when' or ‘with’, and be in the
+form of a sentence with proper grammar.
+
+#### Bad Example
+
+```ruby
+context 'the display name not present' do
+  # ...
+end
+```
+
+#### Good Example
+
+```ruby
+context 'when the display name is not present' do
+  # ...
+end
+```
+
+### `it` descriptions
+
+`it` block descriptions should never end with a conditional. This is a code
+smell that the `it` most likely needs to be wrapped in a `context`. Also, this happens when the description is too long.
```

Comment:
A verb needs to accompany the subject. In this case "code smell" is the subject and you need to describe what it does (it gives a recommendation on the `it`).
E.g.: "This is a code smell that advices the `it` most likely...".

On the second sentence: we tend to use "Also" at the beginning if we are talking about a longer sentence. In this case, the following sentence sounds better: 
"This also happens when..."

Reply:
About the second sentence:
The one you used is correct. 
You can use "also" at the beginning of a sentence to introduce more information to a previous fact.
It's really a matter of personal preference when writing / speaking. Adding an adverb adds a pause (which is why the comma is needed right after):
Furthermore, ....
Definitely, ...
Surely, ...
Also, ...
Personally, I prefer to avoid that pause if the sentence is short. I'd rather have a more continuous line of thought.
Hope this is more clear; but again, your original sentence was correct. Sorry for the confusion.